src/api/forms/repositories/form-metadata-repository.js

```diff
@@ -6,28 +6,56 @@ import { removeById } from '~/src/api/forms/repositories/helpers.js'
 import { createLogger } from '~/src/helpers/logging/logger.js'
 import { METADATA_COLLECTION_NAME, db } from '~/src/mongo.js'
 
-export const MAX_RESULTS = 500
+export const MAX_RESULTS = 100
 
 const logger = createLogger()
 
 /**
  * Retrieves the list of documents from the database
  */
-export function list() {
-  const coll = /** @satisfies {Collection<Partial<FormMetadataDocument>>} */ (
+export async function listAll() {
+  const coll = /** @type {Collection<Partial<FormMetadataDocument>>} */ (
     db.collection(METADATA_COLLECTION_NAME)
   )
 
   return coll
     .find()
     .sort({
-      updatedAt: -1, // Primary sort: most recently updated documents first (-1 = descending order)
-      'updatedBy.displayName': 1 // Secondary sort: if update times are equal, sort by author name (1 = ascending order)
+      updatedAt: -1
     })
     .limit(MAX_RESULTS)
     .toArray()
 }
 
+/**
+ * Retrieves the list of documents from the database with pagination.
+ * @param {object} paginationOptions - Pagination options.
+ * @param {number} paginationOptions.page - The current page number.
+ * @param {number} paginationOptions.perPage - The number of items per page.
```

Comment:
Should we use your type definition for this?

```suggestion
 * @param {PaginationOptions} paginationOptions - Pagination options
```

Reply:
Yeah spot on. Will update this after merging in the shared types.